src/xdp/xsk.c

```diff
@@ -234,11 +234,34 @@ XskSignalReadyIo(
 
     KeAcquireSpinLock(&Xsk->Lock, &OldIrql);
     if ((Xsk->IoWaitFlags & ReadyFlags) != 0) {
+        Xsk->IoWaitFlags &= ~ReadyFlags;
```

Comment:
Left a comment elsewhere, but re-commenting that you can't clear bits here because non-zero bits prevent multiple waits. If this clears the final bit, then another waiter can arrive and restart the wait while the original thread is still the "exclusive" waiter.